From 1b1055c2783884c40968e88c65b9dd3c78b033bf Mon Sep 17 00:00:00 2001
From: Mohammed Hassan <mohammed.hassan@jolla.com>
Date: Thu, 2 Apr 2015 17:13:16 +0300
Subject: [PATCH 2/2] downgrade automake required to 1.11

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1963873..f8966af 100644
--- a/configure.ac
+++ b/configure.ac
@@ -12,7 +12,7 @@ AC_CANONICAL_TARGET
 
 AC_CONFIG_SRCDIR([agent/agent.c])
 AC_CONFIG_HEADERS([config.h])
-AM_INIT_AUTOMAKE([1.12 -Wall -Wno-portability subdir-objects])
+AM_INIT_AUTOMAKE([1.11 -Wall -Wno-portability subdir-objects])
 
 AC_CONFIG_FILES([
 	Makefile
-- 
2.1.4

