diff --git a/autogen.sh b/autogen.sh
index d58c834..2ba29e4 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -2,9 +2,30 @@
 set -e
 
 test -d m4 || mkdir m4
-gtkdocize || exit 1
 
-autoreconf -fi
+# check if gtk-doc is explicitly disabled
+for ag_option in $@
+do
+  case $ag_option in
+    -disable-gtk-doc | --disable-gtk-doc)
+    enable_gtk_doc=no
+  ;;
+  esac
+done
+
+if test x$enable_gtk_doc = xno; then
+    if test -f gtk-doc.make; then :; else
+       echo "EXTRA_DIST = missing-gtk-doc" > gtk-doc.make
+    fi
+    echo "WARNING: You have disabled gtk-doc."
+    echo "         As a result, you will not be able to generate the API"
+    echo "         documentation and 'make dist' will not work."
+    echo
+else
+    gtkdocize || exit $?
+fi
+
+autoreconf -fi -I m4
 
 run_configure=true
 for arg in $*; do
