diff --git a/autogen.sh b/autogen.sh
index b6efba6..8d91eaa 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -22,9 +22,31 @@ if [ "$#" = 0 -a "x$NOCONFIGURE" = "x" ]; then
 	echo "" >&2
 fi
 
+# check if gtk-doc is explicitly disabled
+for ag_option in $@
+do
+  case $ag_option in
+    -disable-gtk-doc | --disable-gtk-doc)
+    enable_gtk_doc=no
+  ;;
+  esac
+done
+
+if test x$enable_gtk_doc = xno; then
+    if test -f gtk-doc.make; then :; else
+       echo "EXTRA_DIST = missing-gtk-doc" > gtk-doc.make
+    fi
+    echo "WARNING: You have disabled gtk-doc."
+    echo "         As a result, you will not be able to generate the API"
+    echo "         documentation and 'make dist' will not work."
+    echo
+else
+    gtkdocize --copy || exit 1
+fi
+
+
 aclocal --install || exit 1
-gtkdocize --copy || exit 1
-autoreconf --verbose --force --install || exit 1
+autoreconf --verbose --force --install -I m4 || exit 1
 
 cd "$olddir"
 if [ "$NOCONFIGURE" = "" ]; then
