diff --git a/tests/meson.build b/tests/meson.build
index e238a92..e05f2bc 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -40,6 +40,8 @@ endif
   ]
 endif
 
+nemo_tests_bin = '/opt/tests/libnice/bin'
+
 foreach tname : nice_tests
   if tname.startswith('test-io-stream') or tname.startswith('test-send-recv') or tname == 'test-bytestream-tcp'
     extra_src = ['test-io-stream-common.c']
@@ -52,7 +54,7 @@ foreach tname : nice_tests
     include_directories: nice_incs,
     dependencies: [nice_deps, libm],
     link_with: [libagent, libstun, libsocket, librandom],
-    install: false)
+    install: true, install_dir: nemo_tests_bin)
   set_variable(tname.underscorify(), exe)
   test(tname, exe)
 
@@ -60,7 +62,7 @@ foreach tname : nice_tests
     wrapper_exe = executable ('nice-test-fullmode-with-stun',
 			      'test-fullmode-with-stun.c',
 			      dependencies: gio_deps,
-			      install: false)
+			      install: true, install_dir: nemo_tests_bin)
     test('test-fullmode-with-stun', wrapper_exe,
 	 args: [stund_exe, test_fullmode],
 	 is_parallel: false,
@@ -78,7 +80,7 @@ if gst_dep.found()
       include_directories: nice_incs,
       dependencies: [nice_deps, gst_check, libm],
       link_with: libnice,
-      install: false)
+      install: true, install_dir: nemo_tests_bin)
     gst_env = environment()
     gst_env.append('GST_PLUGIN_PATH_1_0', join_paths(meson.current_build_dir(), '..', 'gst'))
     test('test-gstreamer', exe, env: gst_env)
