diff -Naur libnice-0.1.4/tests/Makefile.am libnice-0.1.4-install-nemo-tests/tests/Makefile.am
--- libnice-0.1.4/tests/Makefile.am	2012-09-12 18:33:35.000000000 +0000
+++ libnice-0.1.4-install-nemo-tests/tests/Makefile.am	2013-05-02 12:52:37.000000000 +0000
@@ -70,3 +70,14 @@
 all-local:
 	chmod a+x $(srcdir)/check-test-fullmode-with-stun.sh
 	chmod a+x $(srcdir)/test-pseudotcp-random.sh
+
+nemotestsbindir = /opt/tests/libnice/bin
+nemotestsbin_PROGRAMS = $(check_PROGRAMS)
+nemotestsbin_SCRIPTS = $(dist_check_SCRIPTS)
+
+libnice-tests.list:
+	$(AM_V_GEN)echo $(check_PROGRAMS) > $@
+
+nemotestsdatadir = /opt/tests/libnice/data
+nobase_nemotestsdata_DATA = libnice-tests.list
+
