diff --git a/tests/Makefile.am b/tests/Makefile.am
index 84c37be..0d61a34 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -114,3 +114,13 @@ all-local:
 	chmod a+x $(srcdir)/test-pseudotcp-random.sh
 
 EXTRA_DIST = libnice.supp
+
+nemotestsbindir = /opt/tests/libnice/bin
+nemotestsbin_PROGRAMS = $(check_PROGRAMS)
+nemotestsbin_SCRIPTS = $(dist_check_SCRIPTS)
+
+libnice-tests.list:
+	$(AM_V_GEN)echo $(check_PROGRAMS) > $@
+
+nemotestsdatadir = /opt/tests/libnice/data
+nobase_nemotestsdata_DATA = libnice-tests.list
