diff --git a/tests/Makefile.am b/tests/Makefile.am
index 473e5e0..3ea1985 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -150,3 +150,13 @@ EXTRA_DIST = \
 	libnice.supp \
 	meson.build \
 	test-fullmode-with-stun.c
+
+nemotestsbindir = /opt/tests/libnice/bin
+nemotestsbin_PROGRAMS = $(check_PROGRAMS)
+nemotestsbin_SCRIPTS = $(dist_check_SCRIPTS)
+
+libnice-tests.list:
+	$(AM_V_GEN)echo $(check_PROGRAMS) > $@
+
+nemotestsdatadir = /opt/tests/libnice/data
+nobase_nemotestsdata_DATA = libnice-tests.list
